src/ppom-controller.ts

```diff
@@ -372,9 +375,15 @@ export class PPOMController extends BaseControllerV2<
       throw Error('Blockaid validation is available only on ethereum mainnet');
     }
     return await this.#ppomMutex.use(async () => {
-      this.#resetPPOM();
-      await this.#maybeUpdatePPOM();
-      this.#ppom = await this.#getPPOM();
+      if (!this.#ppomInitialised) {
+        const { ppomInit } = this.#ppomProvider;
+        await ppomInit('./ppom_bg.wasm');
+        this.#ppomInitialised = true;
+      }
+      if (!this.#ppom) {
+        await this.#maybeUpdatePPOM();
```

Comment:
i think this should be called in all cases

Reply:
If new files are fetched or network changed ppom is reset so `!this.#ppom` will return true.